File: batchbhavna/FirstApplication/app/src/main/java/com/example/firstapplication/SecondActivity.java
package com.example.firstapplication;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.PopupMenu;

import android.os.Bundle;
import android.util.Log;
import android.view.ContextMenu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

public class SecondActivity extends AppCompatActivity {

    ListView listView;
    String[] scientist = {"Newton","Albert","Merry Query","Elon Mask","Villgets","xys","Newton","Albert","Merry Query","Elon Mask","Villgets","xys"};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        listView = findViewById(R.id.listview);

        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(getApplicationContext(), android.R.layout.simple_spinner_item,scientist);
        arrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        listView.setAdapter(arrayAdapter);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                PopupMenu popupMenu = new PopupMenu(SecondActivity.this,view);
                popupMenu.getMenuInflater().inflate(R.menu.popmenu,popupMenu.getMenu());


                popupMenu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
                    @Override
                    public boolean onMenuItemClick(MenuItem item) {
                        int id = item.getItemId();
                        if(id == R.id.pospupone)
                        {
                            Toast.makeText(SecondActivity.this, "clciked "+item.getTitle(), Toast.LENGTH_SHORT).show();
                        }else if(id == R.id.popseconditem)
                        {
                            Toast.makeText(SecondActivity.this, "clicked "+item.getTitle(), Toast.LENGTH_SHORT).show();
                        }
                        return true;
                    }
                });


                popupMenu.show();
            }
        });
        registerForContextMenu(listView);



    }

    @Override
    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        super.onCreateContextMenu(menu, v, menuInfo);
        getMenuInflater().inflate(R.menu.contextmeny,menu);
    }

    @Override
    public boolean onContextItemSelected(@NonNull MenuItem item) {
        int id = item.getItemId();
        if(id == R.id.contextfirst)
        {
           int result = sum(23,56);
            Toast.makeText(this, "sum is => "+result, Toast.LENGTH_SHORT).show();
        }else if(id == R.id.ContextSecond){
            int result = multi(23,56);
            Toast.makeText(this, "multi is =>"+result, Toast.LENGTH_SHORT).show();
        }
        return super.onContextItemSelected(item);
    }

    private int sum(int a ,int b){
        int sum = a+b;
     return sum;
    }

    private int multi(int a,int b)
    {
        int multi = a*b;
        return multi;
    }
}

File: 4pmclass/MenusinKotlin/app/src/main/java/com/example/menusinkotlin/SecondActivity.kt
package com.example.menusinkotlin

import android.os.Bundle
import android.view.ContextMenu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.PopupMenu

class SecondActivity : AppCompatActivity() {
    lateinit var listview:ListView
    var namelis = arrayListOf<String>("Albert","Newtor","Merry query","Elon mask","Dev","Aditi","Xyz","Albert","Newtor","Merry query","Elon mask","Dev","Aditi","Xyz")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        listview = findViewById(R.id.listview)

        var adapter :ArrayAdapter<String> = ArrayAdapter(applicationContext,android.R.layout.simple_spinner_item,namelis)

        adapter.setDropDownViewResource(android.R.layout.simple_spinner_item)

        listview.adapter = adapter

 listview.setOnItemClickListener { parent, view, position, id ->
   //  Toast.makeText(applicationContext,"name => ${namelis[position]}",Toast.LENGTH_SHORT).show()
var popupMenu:PopupMenu = PopupMenu(this@SecondActivity,view)
     popupMenu.menuInflater.inflate(R.menu.popupmenufile,popupMenu.menu)

     popupMenu.setOnMenuItemClickListener { item ->
         var id = item.itemId
         if(id == R.id.popupone)
         {
             Toast.makeText(applicationContext,"onclck "+item.title,Toast.LENGTH_SHORT).show()
         }else if(id == R.id.popuptwo)
         {
             Toast.makeText(applicationContext,"onclck "+item.title,Toast.LENGTH_SHORT).show()
         }

         true
     }
     popupMenu.show()

 }

        registerForContextMenu(listview)

    }

    override fun onCreateContextMenu(menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?)  {
        super.onCreateContextMenu(menu, v, menuInfo)

        var menuiflater : MenuInflater = menuInflater
        menuiflater.inflate(R.menu.contextmenu,menu)

    }

    override fun onContextItemSelected(item: MenuItem): Boolean {

        var id = item.itemId
        if(id == R.id.contextfirts)
        {
          var result =  sum(23,45)
            Toast.makeText(applicationContext," sum is => $result",Toast.LENGTH_SHORT).show()

        }else if(id == R.id.contextsecond)
        {
           var result : Int = multi(34,56)
            Toast.makeText(applicationContext," multi is => $result ",Toast.LENGTH_SHORT).show()
        }
        return true
    }

    private fun multi(i: Int, i1: Int): Int {
      return i*i1
    }


    private fun sum(i: Int, i1: Int) : Int {
              return  i+i1
    }

}